File: workflow_shell/git/amend.py
from ..util import run


command_string = "ga"
command_help = "Amends the most recent commit to ensure that it has the correct author and an updated timestamp"
short_help = "Git - amends recent commit author and timestamp"


def main(*args, **kwargs):
    run(["git", "commit", "--amend", "--reset-author", "--no-edit"])

File: src/commands/git/Amend.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';

const argumentss: Argument[] = [];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string = 'Amends the most recent commit to ensure that it has the correct author and an updated timestamp';
  alias: string = 'ga';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await executeCommand('git', ['commit', '--amend', '--reset-author', '--no-edit']);
    });
  }
}
